### DISCUSSION TITLE
Hello! I'm new to coding websites and I want to learn how to save responses for my lost and found website.

### DISCUSSION THREAD

**Discussion by Karl-Too** [community/community#142508] [Programming Help]
### Body

Hellooo! I'm currently doing research for my school and I need some help on how to setup my website. 

I've already done some of the basics: 
- static web app
- storage accounts
- sql database/server
- pretty much the design of the website

how do I make it so that responses save when they click submit here:
![image](https://github.com/user-attachments/assets/d275ac94-d061-447b-af4c-4de2dc062ed0)

and it shows to the dashboard and database. Thank you for ur responses! srry for this bad thread design

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hi @Karl-Too, thanks for being a part of the GitHub Community!

Have you started any API routing? You'll need to send a POST request to save the responses to your database.

You are more likely to get a useful response if you share code snippets or a link to your repo. 😄

**Reply by Karl-Too** [reply]
Hello @ebndev, thank you for welcoming me here!

I'm currently using Microsoft Azure for it, is the API routing like those FUNCTION APP in Azure? Can you teach me how to do those API routing? I've been trying to deploy this SQL Database but output terminal says no HTTP Trigger found. Is this the first step or am I doing something wrong here? Thank you for your response!

https://github.com/Karl-Too/ANIMOLOST

**Comment by Karl-Too** [comment]
Hello Everyone, I still need help!

**Comment by DhareshP** [comment]
design an api for POST request, when you're passing something in the input box and click on submit it will save your response in the database

**Reply by Karl-Too** [reply]
Hello @DhareshP  Thanks for responding, I've done a code for the API post request like save entries. I'm testing it in POSTMAN and it says 500 Internal Error?  I'm not so familiar on how to code JavaScript so I asked AI's for some help but it still says 500 Internal Error. 


[the save query code.txt](https://github.com/user-attachments/files/17578594/the.save.query.code.txt)

**Reply by DhareshP** [reply]
your ui code is correct. check the request body you're passing, does it align with the parameter's you're considering. Can you share your backend code? the issue is from the server side, did you check, is your storage account is accessible?
use try catch or check the azure logs, you might find the specific error message

**Reply by Karl-Too** [reply]
Hello @DhareshP, could you tell me about what should be present in the backend code? And how do I know if the storage account is accesible? 

Here are all of the codes that I have done:
[index html.txt](https://github.com/user-attachments/files/17586678/index.html.txt)
[get entry code.txt](https://github.com/user-attachments/files/17586679/get.entry.code.txt)
[the save query code.txt](https://github.com/user-attachments/files/17586680/the.save.query.code.txt)

**Reply by DhareshP** [reply]
use jquery or axios, the files you share are unaccessable, share images/screenshots

example
to fetch an api

fetch('Api_address')
  .then(response => {
    if (response.ok) {
      return response.json(); // Parse the response data as JSON
    } else {
      throw new Error('API request failed');
    }
  })
  .then(data => {
    // Process the response data here
    console.log(data); // Example: Logging the data to the console
  })
  .catch(error => {
    // Handle any errors here
    console.error(error); // Example: Logging the error to the console
  });

in your frontend file import axios or jquery whatever youre using
here is an example,

import axios from 'axios';

axios.get('APIURL')
    .then(response => {
        // Access the response data
        const responseData = response.data;
        // Process the response data here
    })
    .catch(error => {
        // Handle any errors
    });

**Reply by Karl-Too** [reply]
[index html.txt](https://github.com/user-attachments/files/17587107/index.html.txt)
[get entry code.txt](https://github.com/user-attachments/files/17587109/get.entry.code.txt)
[the save query code.txt](https://github.com/user-attachments/files/17587110/the.save.query.code.txt)

My bad, I forgot to save them, they should work now!
